BattleShipsClient/BattleShipsClient/Program.cs

```diff
@@ -11,12 +12,36 @@ static class Program
         /// <summary>
         /// The main entry point for the application.
         /// </summary>
+        
         [STAThread]
         static void Main()
         {
             Application.EnableVisualStyles();
             Application.SetCompatibleTextRenderingDefault(false);
-            Application.Run(new Form1());
+            Application.Run(new Menu());
+            //Application.Run(new Form1());
+        }
+        static string LogName;
+        static object LogLock = new object();
+        public static void Log(string message)
+        {
+            lock (LogLock)
+            {
+                using (StreamWriter swAppend = File.AppendText(LogName))
+                {
+                    swAppend.WriteLine("[" + DateTime.Now.Hour.ToString() + ":" + DateTime.Now.Minute.ToString() + ":" + DateTime.Now.Second.ToString() + "]" + " - " + message);
+                }
+            }
+        }
+        public static void MakeLog()
+        {
+            if (!Directory.Exists("Logs/"))
+            {
+                Directory.CreateDirectory("Logs");
+            }
+            LogName = $"logs/" + DateTime.Today.Day.ToString() + " " + DateTime.Today.Month.ToString() + " & " + DateTime.Now.Hour.ToString() + ";" + DateTime.Now.Minute.ToString() + ".txt";
+            StreamWriter swNew = File.CreateText(LogName);
+            swNew.Close();
```

Comment:
Through use of a `using(StreamWriter swNew = File.CreateText(LogName))` statement
This means that if an error occurs, the stream is closed.
Because your current code causes a memory leak if an error occurs before .Close() is called